### PR TITLE
build: create tdd performance test script

### PR DESCRIPTION
**Pull request by josephperrott**
This script automates the steps from Pawel's "TDD workflow productivity" doc for us to better work to track the our TDD turnaround time.

### CODE REVIEW

File: scripts/local-dev/test-tdd.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Package being tested for turnaround time.
+test_package="//packages/core/test/acceptance:acceptance"
+# Bazel test command
+test_command="yarn -s bazel test ${test_package} --define=compile=aot > /dev/null 2>&1"
+echo "Testing ${test_package}"
+
+# Set up a clean bazel environment
+echo "Running bazel clean and restart bazel server"
+#yarn -s bazel clean --expunge > /dev/null 2>&1
```
Comment (filipesilva) [inline scripts/local-dev/test-tdd.sh]: Should this be commented out?

File: scripts/local-dev/test-tdd.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+yarn -s bazel info > /dev/null 2>&1
+
+# Perform initial build/test to fill the cache
+echo "Testing acceptance target"
+first_run=$(/usr/bin/time -f "%e" sh -c "$test_command" 2>&1 1>/dev/null)
+echo "  Initial build time: ${bold}$first_run seconds${normal}"
+
+# Make a change to invalidate cache
+echo "Adding empty new line to attributes_spec.ts file"
+echo >> ../../packages/core/test/acceptance/attributes_spec.ts
```
Comment (IgorMinar) [inline scripts/local-dev/test-tdd.sh]: will the new line be preserved in the emitted js/d.ts files? I would feel more confident in this if we appended `console.log('hello")` to the file.

File: scripts/local-dev/test-tdd.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Short cuts to set output as bold and normal
+bold=$(tput bold)
+normal=$(tput sgr0)
+
+# Package being tested for turnaround time.
+test_package="//packages/core/test/acceptance:acceptance"
+# Bazel test command
+test_command="yarn -s bazel test ${test_package} --define=compile=aot > /dev/null 2>&1"
```
Comment (IgorMinar) [inline scripts/local-dev/test-tdd.sh]:
```suggestion
test_command="yarn -s bazel test ${test_package} --config=ivy > /dev/null 2>&1"
```

File: scripts/local-dev/test-tdd.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Short cuts to set output as bold and normal
```
Comment (IgorMinar) [inline scripts/local-dev/test-tdd.sh]:
```suggestion
# Shortcuts to set output as bold and normal
```

**Review by IgorMinar — CHANGES_REQUESTED** [review]
could we use shelljs for this script rather than bash? I've been slowly switching over most of our bash scripts to shelljs because it's so much easier to read and maintain. It's also windows friendly.

File: scripts/local-dev/test-tdd.sh (hunk 2 above)
Comment (josephperrott) [reply scripts/local-dev/test-tdd.sh]: The new line is actually preserved, but its much more obvious with the `console.log`

**Pull request closed without merging**